include/IrrCompileConfig.h

```diff
@@ -4,6 +4,7 @@
 
 #ifndef __IRR_COMPILE_CONFIG_H_INCLUDED__
 #define __IRR_COMPILE_CONFIG_H_INCLUDED__
+#define _ENABLE_EXTENDED_ALIGNED_STORAGE
```

Comment:
wrap this in a test for MSVC compiler (only enable if MSVC detected, we have plenty of these tests in the engine headers, so just copy one)